sympy/polys/numberfields.py

```diff
@@ -450,6 +451,40 @@ def _minpoly_cos(ex, x):
     raise NotAlgebraic("%s doesn't seem to be an algebraic element" % ex)
 
 
+def _minpoly_tan(ex):
+    """
+    Returns the minimal polynomial of ``tan(ex)``
+    see https://github.com/sympy/sympy/issues/21430
+    """
+    x = Symbol('x', real=True)
+    c, a = ex.args[0].as_coeff_Mul()
+    if a is pi:
+        if c.is_rational:
+            c = c * 2
+            n = int(c.q)
+            d = [1] * (n + 1)
+            a = 1
+            for k in range(1, n + 1):
+                a = (a * (n - k + 1))//k
+                if k % 4 == 2 or k % 4 == 3:
+                    d[k] = -a
+                else:
+                    d[k] = a
+
+            if c.p % 2 == 0:
+                coeff = {j : d[j] for j in range(1, n+1, 2)}
+            else:
+                coeff = {j : d[j] for j in range(0, n+1, 2)}
+
+            a = [x**i*coeff[i] for i in coeff.keys()]
```

Comment:
A single list of terms should suffice. Something like this in the odd case:
```
        terms = []
        a = 1
        for k in range(0, n+1, 2):
            terms.append(a*x**k)
            a = -(a*(n-k-1)*(n-k)) // ((k+1)*(k+2))
```

Reply:
Sure, there was unnecessary memory used.
Kindly review the latest commit.
Thanks.